src/library_webgl.js

```diff
@@ -632,6 +617,21 @@ var LibraryGL = {
       } else {
 #endif
 
+#if MIN_SAFARI_VERSION != TARGET_NOT_SUPPORTED && GL_WORKAROUND_SAFARI_GETCONTEXT_BUG
+      // BUG: Workaround Safari WebGL issue: After successfully acquiring WebGL context on a canvas,
+      // calling .getContext() will always return that context independent of which 'webgl' or 'webgl2'
+      // context version was passed. See https://bugs.webkit.org/show_bug.cgi?id=222758 and
+      // https://github.com/emscripten-core/emscripten/issues/13295.
+      // TODO: Once the bug is fixed and shipped in Safari, adjust the Safari version field in above check.
+      if (!canvas.getContextSafariWebGL2Fixed) {
+        canvas.getContextSafariWebGL2Fixed = canvas.getContext;
+        canvas.getContext = function(ver, attrs) {
+          var gl = canvas.getContextSafariWebGL2Fixed(ver, attrs);
+          return ((ver == 'webgl') == (gl instanceof WebGLRenderingContext)) ? gl : null;
+        }
+      }
+#endif
```

Comment:
I'm missing something here. This moves code that uses `canvas`. But that variable is not change in any of the code we skipped over. How is the change noticeable?

Reply:
I can't figure out how to make a link to the relevant lines but look at lines (post change) 606 to 618 in `library_webgl.js`. Line 610 has `if (Module['preinitializedWebGLContext']) {` and line 617 just before the moved code has the matching `} else {`. So the code is no longer executed if there a pre-initialized context is being used.

Reply:
Thanks, I see, so it's in the `else`. The lack of indentation confused me. (But it's correct to not be indented, given that the if is ifdefed.)